### PR TITLE
Use FromStr to implement MemInfo and Stat parsing

### PR DESCRIPTION
**Pull request by danieldulaney**
TLDR:
- Break out `MemInfo` and `Stat` parsing into the `FromStr` trait
- Add tests to verify that it still works

Details:

I ran into a need to parse meminfo and stat files that are copied out of `/proc` on different machines. A good way to handle that is by breaking up the reading and parsing parts of the library into different functions. The `FromStr` trait seems like a good option here for where to put the parsing code -- it gives us the `parse` method for free.

This PR does that for `MemInfo` and `Stat`. There look like a lot of line changes, but most of it is just moving the parsing code into `FromStr::from_str`. Also, I added integration tests based on the existing meminfo/stat samples in comments in the original lib.rs, as well as new ones based on my laptop's current `/proc` folder, and based on an empty string.

### CODE REVIEW

**Comment by nathansizemore** [comment]
Seems legit, thanks!

**Comment by nathansizemore** [comment]
FYI 0.3.0 has been published in cargo.
